specs/latest/2.0/index.html

```diff
@@ -3192,6 +3192,37 @@ <h2 class="no-toc">Table of contents</h2>
         <code>FLOAT_32_UNSIGNED_INT_24_8_REV</code>, generates an <code>INVALID_ENUM</code>.
     </p>
 
+    <h3>VertexAttrib function must match shader attribute type</h3>
+
+    <p>
+        If <code>vertexAttrib{1234}f</code> or <code>vertexAttrib{1234}fv</code> is used and the base
+        type of the shader at slot <code>index</code> is not floating-point (e.g. is signed or unsigned
+        integer), generates an <code>INVALID_OPERATION</code> error.
+    </p>
+    <p>
+        If <code>vertexAttribI4ui</code> or <code>vertexAttribI4uiv</code> is used and the base type of
+        the shader attribute at slot <code>index</code> is floating-point or signed integer,
+        generates an <code>INVALID_OPERATION</code> error.
+    </p>
+    <p>
+        If <code>vertexAttribI4i</code> or <code>vertexAttribI4iv</code> is used and the base type of
+        the shader attribute at slot <code>index</code> is floating-point or unsigned integer,
+        generates an <code>INVALID_OPERATION</code> error.
+    </p>
+
+    <h3>Transform Feedback Primitive Capture</h3>
+
+    <p>
+        If any output variable is specified to be streamed to a buffer object but not actually
+        written by a vertex shader, the buffer remains untouched.
+    </p>
+
+    <h3>gl_FragDepth</h3>
+
+    <p>
+        If a fragment shader statically assigns a value to <code>gl_FragDepth</code>, for any fragment
+        where statements assigning a value to gl_FragDepth are not executed, the value 0 is used.
+    </p>
```

Comment:
Could you point to the ES 3.0 spec text that leaves this behavior undefined? How will this be handled? By assigning to gl_FragDepth at the beginning of the shader, and then leaving the user's shader untouched (so that if gl_FragDepth is conditionally assigned, it gets the 0 value assigned at the beginning)?